### PR TITLE
[WIP] fix bugs

### PR DESCRIPTION
**Pull request by qindazhu**
Will continue working on this to fix other bugs.

### CODE REVIEW

File: k2/python/csrc/torch/torch_util.cu [hunk 1]
```diff
@@ -80,6 +80,7 @@ torch::Tensor ToTensor(Array1<Arc> &array) {
   std::vector<int64_t> sizes = {array.Dim(), 4};  // [num_rows, num_cols]
   std::vector<int64_t> strides = {4, 1};          // in number of elements
   auto options = torch::device(device).dtype(scalar_type);
+  if (array.Dim() == 0) return torch::empty((0, 4), options);
```
Comment (csukuangfj) [inline k2/python/csrc/torch/torch_util.cu:83]: `(0, 4)` should be changed to `{0, 4}`.

Comment (csukuangfj) [reply k2/python/csrc/torch/torch_util.cu:83]: `(0, 4)` is a comma expression in C++.
So `torch::empty((0, 4), options);` is equivalent to `torch::empty(4, options);`

**Comment by danpovey** [comment]
Thanks!!

File: k2/python/csrc/torch/torch_util.cu (hunk 1 above)
Comment (qindazhu) [reply k2/python/csrc/torch/torch_util.cu:83]: Oh, yes, and @csukuangfj can you kindly help to check other code of those convertion between k2 and torch, I think we should add tests for empty cases

Comment (csukuangfj) [reply k2/python/csrc/torch/torch_util.cu:83]: Sure, I'd like to do it.

Comment (qindazhu) [reply k2/python/csrc/torch/torch_util.cu]: Updated.
